api/envoy/api/v2/core/protocol.proto

```diff
@@ -19,6 +19,12 @@ option (udpa.annotations.file_migrate).move_to_package = "envoy.config.core.v3al
 message TcpProtocolOptions {
 }
 
+message UpstreamHttpProtocolOptions {
+  // Set transport socket SNI for new upstream connections based on the downstream HTTP
+  // host/authority header, as seen by the router filter
```

Comment:
End with period, also ref link to the router filter docs as I asked previously. This feature also needs a release note which should link to this field, and possibly a mention in the router/HTTP arch overview/config documentation.